services/ux-backend/handlers/featureflags/handler.go

```diff
@@ -0,0 +1,214 @@
+		return false, fmt.Errorf("failed to list NooBaa CRs: %w", err)
+	}
+
+	if len(noobaaList.Items) > 0 {
+		klog.Info("Found NooBaa CRs in namespace:", namespace, "count:", len(noobaaList.Items))
+		return true, nil
+	}
+
+	klog.Info("No NooBaa CRs found in namespace:", namespace)
+	return false, nil
+}
+
+func checkNoobaaStorageClassExists(ctx context.Context, client ctrlclient.Client, namespace string) (bool, error) {
+	klog.Info("Checking for NooBaa StorageClass existence")
+	
+	provisioner := fmt.Sprintf("%s.noobaa.io/obc", namespace)
+	storageClassList := &storagev1.StorageClassList{}
+	if err := client.List(ctx, storageClassList, ctrlclient.MatchingFields{"provisioner": provisioner}, ctrlclient.Limit(1)); err != nil {
+		return false, fmt.Errorf("failed to list StorageClasses by provisioner index: %w", err)
+	}
+	
+	if len(storageClassList.Items) > 0 {
+		klog.Info("Found NooBaa StorageClass:", storageClassList.Items[0].Name, "with provisioner:", storageClassList.Items[0].Provisioner)
+		return true, nil
+	}
+	
+	klog.Info("No NooBaa StorageClasses found")
+	return false, nil
+}
+
+func checkRGWFlag(ctx context.Context, client ctrlclient.Client, namespace string) (bool, error) {
+	cephObjectStoreExists, err := checkCephObjectStoreExists(ctx, client)
+	if err != nil {
+		return false, fmt.Errorf("failed to check CephObjectStore: %w", err)
+	}
+	if !cephObjectStoreExists {
+		return false, nil
+	}
+
+	storageClassExists, err := checkRGWStorageClassExists(ctx, client, namespace)
+	if err != nil {
+		return false, fmt.Errorf("failed to check RGW StorageClass: %w", err)
+	}
+
+	return storageClassExists, nil
+}
+
+func checkCephObjectStoreExists(ctx context.Context, client ctrlclient.Client) (bool, error) {
+	klog.Info("Checking CephObjectStore existence")
+
+	cephObjectStoreList := &cephv1.CephObjectStoreList{}
+	if err := client.List(ctx, cephObjectStoreList); err != nil {
+		return false, fmt.Errorf("failed to list CephObjectStores: %w", err)
+	}
+
+	if len(cephObjectStoreList.Items) > 0 {
+		klog.Info("Found CephObjectStores, count:", len(cephObjectStoreList.Items))
+		return true, nil
+	}
+
+	klog.Info("No CephObjectStores found")
+	return false, nil
+}
+
+func checkRGWStorageClassExists(ctx context.Context, client ctrlclient.Client, namespace string) (bool, error) {
+	klog.Info("Checking for RGW StorageClass existence")
+	
+	provisioner := fmt.Sprintf("%s.ceph.rook.io/bucket", namespace)
+	storageClassList := &storagev1.StorageClassList{}
+	if err := client.List(ctx, storageClassList, ctrlclient.MatchingFields{"provisioner": provisioner}, ctrlclient.Limit(1)); err != nil {
+		return false, fmt.Errorf("failed to list StorageClasses by provisioner index: %w", err)
+	}
+	
+	if len(storageClassList.Items) > 0 {
+		klog.Info("Found RGW StorageClass:", storageClassList.Items[0].Name, "with provisioner:", storageClassList.Items[0].Provisioner)
+		return true, nil
+	}
+	
+	klog.Info("No RGW StorageClasses found")
+	return false, nil
+}
```

Comment:
I think this comment was not addressed. I still see chaining in the functions that is increasing the call depth. I think you should let the reviewer resolve the comments https://github.com/red-hat-storage/ocs-operator/pull/3536#discussion_r2468005832

Reply:
We shouldn't create unnecessary functions if they aren't required. In this case, you are concerned about whether Noobaa/CephRGW is available or not. I don't think we need to split the logic for CR and storageClass into two different functions. I would argue that we aren't benefiting from it; instead, it's making the code less readable and making me jump from one line to another to see what the code is trying to achieve.

Reply:
I don't want to force, you are nack'ng the change. let it be, I don't have anymore supporting arguments.

Reply:
Resolving this comment earlier was a mistake on my end.